pyomo/core/expr/compare.py

```diff
@@ -230,10 +230,14 @@ def assertExpressionsEqual(test, a, b, include_named_exprs=True, places=None):
         test.assertEqual(len(prefix_a), len(prefix_b))
         for _a, _b in zip(prefix_a, prefix_b):
             test.assertIs(_a.__class__, _b.__class__)
-            if places is None:
-                test.assertEqual(_a, _b)
+            # If _a is nan, check _b is nan
+            if _a != _a:
+                test.assertTrue(_b != _b)
             else:
-                test.assertAlmostEqual(_a, _b, places=places)
+                if places is None:
+                    test.assertEqual(_a, _b)
+                else:
+                    test.assertAlmostEqual(_a, _b, places=places)
```

Comment:
Instead of this, should we just have `places` default to the default value in `assertAlmostEqual`?

Reply:
There are a few tests in `contrib.piecewise` where I had to reduce it because of numerical stuff. So that's why there's an argument for it.